src/lib/context/InternalFormContext.tsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext } from "react";
+import { FieldValues } from "react-hook-form";
+
+export interface InternalFormContextProps<T extends FieldValues> {
+  requiredFields?: (keyof T)[];
+}
+
+export interface InternalFormContextProviderProps<T extends FieldValues> extends Pick<InternalFormContextProps<T>, "requiredFields"> {
+  children: React.ReactNode;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const InternalFormContext = createContext<InternalFormContextProps<any> | null>(null);
+
+export const InternalFormProvider = <T extends FieldValues>(props: InternalFormContextProviderProps<T>) => {
```

Comment:
I suggest renaming this to something more specific like `RequiredFieldsContext` otherwise it's confusing and not clear what this context actually does.

Reply:
@manni497 

I see your point. I would say it would be easy to refactor it to something more generic if we are adding more properties that we would like to keep track off. But for now I don't have anything in mind that we could add to the context. 

@neotob  What's your opinion? 

In the end I have no hard feelings about the naming, both is acceptable for me.

Reply:
i see both your points, in the end for just now it would probably be more accurate to name it specifically, but on the other side, especially if we say we wanna keep it open for more properties in the future, having a specific name will keep others from reusing it and they will maybe even create additional context's...

so for me its fine to go with the current name, especially if we say we want it to be reused in the future, in case there will be possible additional properties.

Reply:
@neotrow @neotob  Name of the context left as it was :D